### ISSUE TITLE
Please update LICENSE.txt to reflect all licenses used

### ISSUE THREAD

**Issue by tflink** [ROCmSoftwarePlatform/MIOpen#2757]
There are several files in the MIOpen repository which have headers which indicate that they are not (just) licensed as MIT.

Please include the text of those licenses in the LICENSE.txt file to make it easier to distribute MIOpen binaries.

* BSD-2-Clause
  * driver/mloSoftmaxHost.hpp
* BSD-2-Clause and MIT
  * src/include/miopen/mlo_internal.hpp
* Apache-2.0 and MIT
  * src/include/miopen/kernel_cache.hpp
  * src/kernel_cache.cpp
* Public Domain (and MIT)
  * src/md5.cpp

**Comment by ppanchad-amd** [comment]
@tflink Can we close this ticket? Thanks!

**Comment by tflink** [comment]
@ppanchad-amd I don't understand why you're asking me, it's not my project. While reviewing MIOpen for packaging in Fedora, we found what we believe to be a discrepancy between the listed software license and the actual state of the files being distributed. In the interest of being good citizens, Fedora's policy is to flag any licensing issued found with the upstream project. I went a slight bit further and submitted a pull request to help fix the discrepancy.

As near as I can tell, the underlying issue has not been resolved. I did what I felt I needed to do in reporting the issue and offering a fix. If the powers that be disagree that this is even an issue or choose to ignore the ticket and PR by closing them without action, so be it.
